refactor/from-ktor-to-retrofit

mohammed-akkad opened the pull request.

mohammed-akkad committed:
update libs versions

mohammed-akkad committed:
update build gradle app

mohammed-akkad committed:
update aflami application

mohammed-akkad committed:
update module in app

mohammed-akkad committed:
update build gradle remote movie

mohammed-akkad committed:
update movie details remote data source

mohammed-akkad committed:
refactor media play button

mohammed-akkad committed:
add get movie video use case

mohammed-akkad committed:
update movie repository

mohammed-akkad committed:
update media details destinations

mohammed-akkad committed:
update media details ui

mohammed-akkad committed:
add web view composable

mohammed-akkad committed:
add service module

mohammed-akkad committed:
add retrofit movie details api service

mohammed-akkad committed:
add movie video in domain

mohammed-akkad committed:
add video web view ui

mohammed-akkad committed:
add movie video dto

mohammed-akkad committed:
update movie details remote data source

mohammed-akkad committed:
update movie mapper

mohammed-akkad committed:
update movie repository

mohammed-akkad committed:
refactor tv show data source

mohammed-akkad committed:
update release info

mohammed-akkad committed:
update movie repository test

Joseph-Sameh-0 reviewed (approved):
Please update from develop

mohammed-akkad committed:
update tv show repository test

mohammed-akkad committed:
Merge remote-tracking branch 'origin/develop' into refactor/from-ktor-to-retrofit

# Conflicts:
#	app/build.gradle.kts
#	app/release-info.txt
#	app/src/main/java/com/paris_2/aflami/AflamiApplication.kt
#	app/src/main/java/com/paris_2/aflami/di/NetworkModule.kt
#	app/src/main/java/com/paris_2/aflami/di/UseCaseModule.kt
#	gradle/libs.versions.toml

mohammed-akkad committed:
fix conflict

mohammed-akkad committed:
update network module

mohammed-akkad committed:
update service module

mohammed-akkad committed:
update aflami application

mohammed-akkad committed:
add media details video use case test

mohammed-akkad committed:
Update MovieRepositoryImplTest.

mohammed-akkad committed:
Update RetrofitTvShowDetailsApiService.

mohammed-akkad committed:
Update TvShowDetailsRemoteDataSourceImpl

mohammed-akkad committed:
Update MovieRepositoryImpl

mohammed-akkad committed:
Update VideoWebViewViewModel

mohammed-akkad committed:
Update TvShowUiState

mohammed-akkad committed:
Update MovieUiState

Joseph-Sameh-0 committed:
Merge branch 'develop' into refactor/from-ktor-to-retrofit

Joseph-Sameh-0 committed:
update build.gradle.kts

mohammed-akkad committed:
update build gradle

mohammed-akkad committed:
add fake data

mohammed-akkad committed:
add media details remote data source test

mohammed-akkad committed:
update build gradle

mohammed-akkad committed:
update build gradle

mohammed-akkad committed:
update build gradle

Joseph-Sameh-0 committed:
Update build.gradle.kts

Joseph-Sameh-0 committed:
Update build.gradle.kts

mohammed-akkad committed:
update movie repository test

mohammed-akkad committed:
Merge remote-tracking branch 'origin/refactor/from-ktor-to-retrofit' into refactor/from-ktor-to-retrofit

Joseph-Sameh-0 committed:
Update build.gradle.kts

mohammed-akkad committed:
update tv show repository test

mohammed-akkad committed:
update build gradle

Joseph-Sameh-0 committed:
Merge branch 'refactor/from-ktor-to-retrofit' of https://github.com/Paris-Squad-S2/Aflami into refactor/from-ktor-to-retrofit

Joseph-Sameh-0 committed:
Revert "update build gradle"

This reverts commit ac10887f45b6a0a235b98b8a127eb01f5258b7d9.

Joseph-Sameh-0 committed:
Refactor: Simplify exception handling in TvShowRepositoryImpl

This commit refactors the exception handling in `TvShowRepositoryImpl` to catch `AflamiException` directly, reducing redundant catch blocks for specific subclasses.

The `safeCall` function now catches `AflamiException` and re-throws it, and catches generic `Exception` to wrap it in a `NetworkException`. This simplifies the error handling logic and makes it more concise.

Joseph-Sameh-0 committed:
Update build.gradle.kts

Joseph-Sameh-0 committed:
Update build.gradle.kts

mohammed-akkad committed:
change search data source to use retrfit instead of ktor

mohammed-akkad committed:
update release info

mohammed-akkad committed:
update network module

mohammed-akkad committed:
update search media repository test

mohammed-akkad committed:
add video web view model test

mohammed-akkad committed:
update video web view model test

mohammed-akkad committed:
Merge remote-tracking branch 'origin/develop' into refactor/from-ktor-to-retrofit

# Conflicts:
#	app/release-info.txt

mohammed-akkad committed:
update release info

mohammed-akkad committed:
update build gradle

Joseph-Sameh-0 committed:
Update Kover minimum code coverage value

The Kover minimum code coverage value in `feature/mediaDetails/mediaDetailsUi/build.gradle.kts` has been temporarily set to 0.

This change is intended to be temporary, with a plan to set a proper minimum value and add tests to achieve it in the future.

Joseph-Sameh-0 committed:
Merge branch 'develop' into refactor/from-ktor-to-retrofit
